src/sql-parser/src/parser.rs

```diff
@@ -3796,7 +3796,17 @@ impl<'a> Parser<'a> {
             MANUAL => ClusterScheduleOptionValue::Manual,
             ON => {
                 self.expect_keyword(REFRESH)?;
-                ClusterScheduleOptionValue::Refresh
+                // Parse optional `(REHYDRATION TIME ESTIMATE ...)`
+                let rehydration_time_estimate = if self.consume_token(&Token::LParen) {
+                    self.expect_keywords(&[REHYDRATION, TIME, ESTIMATE])?;
+                    let _ = self.consume_token(&Token::Eq);
+                    let interval = self.parse_interval_value()?;
+                    self.expect_token(&Token::RParen)?;
+                    Some(interval)
+                } else {
+                    None
+                };
```

Comment:
In theory you should be going through the whole `WithOptionValue` and the `generate_extracted_config!` macro here. I don't blame you for not wanting to bother though. Seems fine for now since there's only a single option, but we'll need to refactor if there's ever a second option we want to support in this position.

Reply:
Ok, I'll keep this in mind if we add more options later.